iambic/plugins/v0_1_0/aws/utils.py

```diff
@@ -3,17 +3,73 @@
 import asyncio
 import re
 from enum import Enum
-from typing import TYPE_CHECKING, Optional, Union
+from typing import TYPE_CHECKING, Any, Optional, Union
 
 import boto3
 from botocore.exceptions import ClientError, NoCredentialsError
 
 from iambic.core.iambic_enum import IambicManaged
 from iambic.core.logger import log
-from iambic.core.utils import aio_wrapper
+from iambic.core.utils import aio_wrapper, is_regex_match
 
 if TYPE_CHECKING:
-    from iambic.plugins.v0_1_0.aws.iambic_plugin import AWSConfig
+    from iambic.plugins.v0_1_0.aws.iambic_plugin import AWSConfig, ImportAction
+
+
+async def process_import_rules(
+    config: AWSConfig,
+    template_type: str,
+    identifier: str,
+    tags: list[dict[str, str]],
+    resource_dict: dict[str, Any],
+) -> list[ImportAction]:
+    actions = []
+
+    for rule in config.import_rules:
+        # Initialize match indicators
+        tag_match = name_match = path_match = template_type_match = False
+
+        # Match by tag
+        if rule.match_tags:
+            for tag in rule.match_tags:
+                for role_tag in tags:
+                    if is_regex_match(tag.key, role_tag.get("key")) and (
```

Comment:
Todo: This was refactored from roles to generic AWS resources, need to rename `role_tag`